warehouse/legacy/api/xmlrpc/views.py

```diff
@@ -108,6 +110,33 @@ def wrapped(context, request):
     return decorator
 
 
+def ratelimit():
+    def decorator(f):
+        def wrapped(context, request):
+            ratelimiter = request.find_service(
+                IRateLimiter, name="xmlrpc.client", context=None
+            )
+            metrics = request.find_service(IMetricsService, context=None)
+            ratelimiter.hit(request.remote_addr)
+            if not ratelimiter.test(request.remote_addr):
+                metrics.increment("warehouse.xmlrpc.ratelimiter.exceeded", tags=[])
+                message = (
+                    "The action could not be performed because there were too "
+                    "many requests by the client."
+                )
+                _resets_in = ratelimiter.resets_in(request.remote_addr)
+                if _resets_in is not None:
+                    _resets_in = int(_resets_in.total_seconds())
+                    message += f" Limit may reset in {_resets_in} seconds."
```

Comment:
```suggestion
                    message += f" Limit will reset in {_resets_in} seconds."
```
Is there a case where the limit won't reset in that time?

Reply:
D'oh, seeing Donald's comment now. Still seems like a weird error message though.

Reply:
The `may` is for the case where we configure a compound rate limit like "1 per second;30 per minute;900 per hour" as supported by our limit library. The resets_in function returns the first limit exceeded (or None) as it is currently implemented.

Reply:
Since we're only using the one identifier (IP address) I think it's probably OK to say "will reset" here.

Reply:
it's not a matter of IP, but the multiple limits that may be enforced.

Reply:
see https://limits.readthedocs.io/en/stable/string-notation.html

Reply:
Got it, didn't know we could compound the limit notation like that. 👍